microprofile/WFLY-12609_MicroProfile_JWT_1_1.adoc

```diff
@@ -0,0 +1,109 @@
+
+However there are a couple of draw backs from the EE Security approach.  A first drawback is this brings in dependencies on JASPI and JACC both of which must be correctly enabled on the server, the `MP-JWT` mechanism is a very trivial light weight mechanism so a lot of overhead is pulled in.  A second draw back is that it should be possible for standard and vendor specific mechanisms to be specified, this brings us additional challenges as many of our mechanisms are native WildFly Elytron mechanisms.  Making use of native mechanisms has it's own complexities as we may need to bridge into using an `@IdentityStore` made available by CDI, alternatively this may require configuration of domains not covered by MicroProfile Config values.  We could consider porting some of our vendor specific mechanisms to EE security but one of the features of our mechanisms is the ability to support multiple mechanisms concurrently which is not completely covered by EE security.  The `MP-JWT` mechanism is also ideally suited to being used in parallel with other authentication mechanisms however we could end up in a situation with a hybrid of native mechanisms and an EE Security mechanism.
+
+Overall these issues can be considered and solved, however I believe the primary motivation for activating MicroProfile JWT is to make use of the `MP-JWT` authentication mechanism so that will be the primary focus of this RFE, depending on time constraints consideration of alternative authentication mechanisms will either take place towards the end of this enhancement or in a follow up enhancement.
+
+=== Non-Requirements
+
+This integration will be making use of WildFly Elytron exclusively, no support will be added for the deprecated PicketBox project.  Having said that the underlying security framework will be transparent for deployments developed according to the specification - the framework only becomes apparent if there is a desire to use vendor specific APIs.
+
+This integration will not be supporting the use of pre-configured SecurityDomains, the general motivation to use MicroProfile JWT is to use an authentication mechanism where identities are self described using the tokens exchanged eliminating the need for configured security resources for the secured endpoint(s).  Use cases could be identified where pre-configured resources add an additional benefit but those would be reviewed under their own enhancements.
+
+It is assumed when working with the MicroProfile specifications that we will be supporting a single deployment, this deployment may contain sub-deployments.  The SecurityDomain of each top level deployment will be isolated from the SecurityDomain of any parallel top level deployment.  Advanced WildFly Elytron features such identity propagation across domains / top level deployments will not be supported.  At a later point we could consider an enhancement to offer more control of the SecurityDomains in use but that is out of scope for this enhancement.
```

Comment:
> At a later point we could consider an enhancement to offer more control of the SecurityDomains in use but that is out of scope for this enhancement.

Meaning WF 20+ ?

Reply:
WF 19+ - but regardless not this enhancement.